text/0080-datetime-extension.md

```diff
@@ -0,0 +1,470 @@
+forbid(
+  principal,
+  action,
+  resource
+) when {
+  context.currentTime > 1580511600 &&
+  context.location.countryOfOrigin == 'GB' &&
+  resource.owner == 'EU'
+}
+```
+
+### Workaround B: Pass results of time checks in the context
+
+Another workaround we have suggested is to simply handle date/time logic _outside_ of Cedar, and pass the results of checks in the `context`. For example, you could pass in fields like  `context.isWorkingHours` or `context.dayOfTheWeek`.
+
+Here are the previous examples rewritten to use additional context.
+
+**Only allow experienced, tenured persons from the Hardware Engineering department to see prototypes.**
+
+```cedar
+permit(
+  principal is User,
+  action == Action::"view",
+  resource in Folder::"device_prototypes"
+)
+when {
+  principal.department == "HardwareEngineering" &&
+  principal.jobLevel >= 10 &&
+  principal.hireDate <= context.minimumHiringDateForAccess
+};
+```
+
+Note: assumes `context.hireDate` and `context.minimumHiringDateForAccess` are `long` values (e.g., Unix time).
+
+**Only allow user "alice" to view JPEG photos for one week after creation.**
+
+```cedar
+permit(
+  principal == User::"alice",
+  action == PhotoOp::"view",
+  resource is Photo
+) when {
+  resource.fileType == "JPEG" &&
+  resource.creationDate >= context.oldestViewableDate
+};
+```
+
+Note: assumes `resource.creationDate` and `context.oldestViewableDate` are `long` values (e.g., Unix time).
+
+**Allow access from a certain IP address only between 9am and 6pm UTC.**
+
+```cedar
+permit(
+  principal,
+  action == Action::"access",
+  resource
+) when {
+  context.srcIp.isInRange(ip("192.168.1.0/24")) &&
+  context.isWorkingHours
+};
+```
+
+**Prevent employees from accessing work documents on the weekend.**
+
+```cedar
+forbid(
+  principal,
+  action == Action::"access",
+  resource is Document
+) when {
+  context.isTheWeekend
+};
+```
+
+**Permit access to a special opportunity for persons born on Leap Day.**
+
+```cedar
+permit(
+  principal,
+  action == Action::"redeem",
+  resource is Prize
+) when {
+  context.isMyBirthday && context.isLeapDay
+};
+```
+
+**Forbid access to EU resources after Brexit**
+
+```cedar
+forbid(
+  principal,
+  action,
+  resource
+) when {
+  context.afterBrexit &&
+  context.location.countryOfOrigin == 'GB' &&
+  resource.owner == 'EU'
+}
+```
+
+## Discussion notes
+
+### Local time
+
+The current proposal has no direct support for local time, or time zones outside of UTC. Fundamentally, adding arbitrary time zone support would complicate the formal model, and ruin the project's SMT-based analysis goals.
+
+Policy writers can use the `datetime.offset()` method to simulate local time. Within the application, time zone data can be used to construct a `duration` that represents the time zone offset in hours and minutes, and provide that through `context` or another entity. Consider the policy below that checks if a principal's local time is between 09:00 and 17:00.
```

Comment:
The discussion here ignores that above, `datetime()` is proposed to actually support a `+/-hhmm` offset.  This is worth mentioning in this discussion; while it doesn't amount to full support for local time, it does allow users to clearly specify times in (say) UTC-5.

Reply:
Updated: https://github.com/strongdm/cedar-rfcs/blob/datetime-rfc/text/0080-datetime-extension.md#local-time